### PR TITLE
CMR-4479: Updated api doc; Added lots of tests.

### PR DESCRIPTION
**Pull request by ygliuvt**

### CODE REVIEW

File: system-int-test/test/cmr/system_int_test/search/service/concept_retrieval_test.clj [hunk 1]
```diff
@@ -21,160 +21,104 @@
...
-                                      {:accept unsupported-mt
-                                       :throw-exceptions true}))]
-        (is (= 400 status))
-        (is (= [(format (str "The mime types specified in the accept header "
-                             "[%s] are not supported.")
-                        unsupported-mt)]
-               errors))))))
+(defn- verify-retrieved-concept
+  "Verify the retrieved concept by checking against the expected service name,
+  which we have deliberately set to be different form each concept revision."
```
Comment (chris-durbin) [inline system-int-test/test/cmr/system_int_test/search/service/concept_retrieval_test.clj]: form typo

File: system-int-test/test/cmr/system_int_test/search/service/concept_retrieval_test.clj [hunk 2]
```diff
@@ -21,160 +21,104 @@
...
+  [concept-id revision-id accept-format expected-service-name]
+  (let [{:keys [status body]} (search/retrieve-concept
+                               concept-id revision-id {:accept accept-format})]
+    (is (= 200 status))
+    (is (=  expected-service-name
```
Comment (chris-durbin) [inline system-int-test/test/cmr/system_int_test/search/service/concept_retrieval_test.clj]: extra space

File: system-int-test/test/cmr/system_int_test/search/service/concept_retrieval_test.clj [hunk 3]
```diff
@@ -21,160 +21,104 @@
...
+            (:Name (json/parse-string body true))))))
+
+(defmulti handle-retrieve-concept-error
+  "Exectue the retrieve concept call with the given parameters and returns the status and errors
```
Comment (chris-durbin) [inline system-int-test/test/cmr/system_int_test/search/service/concept_retrieval_test.clj]: Execute typo

File: system-int-test/test/cmr/system_int_test/search/service/concept_retrieval_test.clj (hunk 1 above)
Comment (ygliuvt) [reply system-int-test/test/cmr/system_int_test/search/service/concept_retrieval_test.clj]: Done.

File: system-int-test/test/cmr/system_int_test/search/service/concept_retrieval_test.clj (hunk 2 above)
Comment (ygliuvt) [reply system-int-test/test/cmr/system_int_test/search/service/concept_retrieval_test.clj]: Done.

**Pull request merged**